shared/types/proto/tx.proto

```diff
@@ -0,0 +1,15 @@
+syntax = "proto3";
```

Comment:
Thought on renaming this file to `transaction_indexer.proto` so it'll contain everything related to this business logic?

Simply `tx.proto` could be unclear to new readers of the coebase.